libsolidity/parsing/Parser.cpp

```diff
@@ -1606,7 +1606,10 @@ ASTPointer<Expression> Parser::parseBinaryExpression(
 		{
 			Token op = m_scanner->currentToken();
 			m_scanner->next();
-			ASTPointer<Expression> right = parseBinaryExpression(precedence + 1);
+			// Parse a**b**c as a**(b**c)
+			ASTPointer<Expression> right = op == Token::Exp ?
+				parseBinaryExpression(precedence) :
+				parseBinaryExpression(precedence + 1);
```

Comment:
Since `EXP` has the highest precedence, i.e., 14, this should work.

Reply:
Is it possible to add this as an assertion here? Also can you put the ternary operator inside the parentheses?

Reply:
I added a static assert. It's a bit ugly though.